pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -893,7 +893,8 @@ type GCPCloudSpec struct {
 type KubevirtCloudSpec struct {
 	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
 
-	Kubeconfig string `json:"kubeconfig,omitempty"`
+	Kubeconfig    string `json:"kubeconfig,omitempty"`
+	CSIKubeConfig string `json:"csiKubeconfig,omitempty"`
```

Comment:
Why is this added here? I don't think I understand the kubevirt related changes in this PR. Are they intentional?

Reply:
This is the only field (at least, the only one I found) that retrieve is enforced through a secret; this means that to make the whole thing work, we'd need to have a cluster from which to get a secret with that value stored. For all the other fields cloud-related, we first try to get the value from the cloudSpec, then if it doesn't exist we try to fetch a secret containing the data. In the image-loader scenario, we must have a field to set because there is no cluster in place to get the secret from (we could use a fake client, but it looks sooo hacky to me). You can see all the credential fetching methods here: https://github.com/kubermatic/kubermatic/blob/8552b8ed98ed2cfb8011b215c2b9d13c00649d28/pkg/resources/credentials.go#L316-L629

Reply:
Okay that actually makes a lot of sense, thank you. @hdurand0710 could you please take a quick look at this PR from a kubevirt perspective? Are there any objections to this (being able to pass the CSI kubeconfig directly instead of relying on a secret)?

Reply:
It makes sense and look good me on a KubeVirt perspective. I've launched the Kubevirt e2e test to be on the safe side.